datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -524,41 +667,24 @@ impl CommonSubexprEliminate {
 /// ```
 ///
 /// where, it is referred once by each `WindowAggr` (total of 2) in the plan.
-struct ConsecutiveWindowExprs {
```

Comment:
I had to remove this struct as `arrays_per_window` is actually a `Vec<Vec<IdArray>>` and now that `Identifier`s in `IdArray`s are references to expressions, it is not possible to store an `Expr` and a reference to the `Expr` in a common struct.

Reply:
That makes sense -- I put it all in a struct initially to try and encapsulate the logic -- I think your changes look good to me